app/models/spree/gateway/mollie_gateway.rb

```diff
@@ -214,8 +214,10 @@ def update_by_mollie_status!(mollie_payment, payment)
         payment.order.update_attributes(:state => 'complete', :completed_at => Time.now)
       when 'canceled', 'expired', 'failed'
         payment.failure! unless payment.failed?
+        payment.order.update_attributes(:state => 'payment', :completed_at => nil)
```

Comment:
Could you add a comment why this is necessary? And note that it's especially important for payments created through the Spree API?

Reply:
Specifically for API's but also for normal website implementations this 'reset' is required to allow the user to select another payment method when their current payment attempt has failed. 

When the order is set to state payment, the payment methods are exposed again and allow for above behaviour